### PR TITLE
fix(iterable): batching order issue

### PR DESCRIPTION
**Pull request by mihir-4116**
## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: src/v0/destinations/iterable/transform.js [hunk 1]
```diff
@@ -166,72 +166,50 @@ const processRouterDest = async (inputs, reqMetadata) => {
...
-            metadata: event.metadata,
-            destination: event.destination,
-          });
-        });
-        return transformedPayloads;
-      } catch (error) {
-        return handleRtTfSingleEventError(event, error, reqMetadata);
-      }
-    }),
+  const groupedEvents = _.groupBy(inputs, (event) => event.message.rudderId);
```
Comment (utsabc) [inline src/v0/destinations/iterable/transform.js:169]: `rudderId` we need to take from metadata

File: src/v0/destinations/iterable/transform.js [hunk 2]
```diff
@@ -166,72 +166,94 @@ const processRouterDest = async (inputs, reqMetadata) => {
...
-            message: response,
-            metadata: event.metadata,
-            destination: event.destination,
-          });
-        });
-        return transformedPayloads;
-      } catch (error) {
-        return handleRtTfSingleEventError(event, error, reqMetadata);
-      }
+  const groupedEvents = _.groupBy(inputs, (event) => event.metadata.rudderId);
```
Comment (utsabc) [inline src/v0/destinations/iterable/transform.js:169]:
```suggestion
  const eventsPerUser = _.groupBy(inputs, (event) => event.metadata.rudderId);
```

File: src/v0/destinations/iterable/transform.js [hunk 3]
```diff
@@ -166,72 +166,94 @@ const processRouterDest = async (inputs, reqMetadata) => {
...
+  const eventsArrays = Object.values(groupedEvents);
```
Comment (utsabc) [inline src/v0/destinations/iterable/transform.js:170]:
```suggestion
  const eachUserEvents = Object.values(groupedEvents);
```

File: src/v0/destinations/iterable/transform.js [hunk 4]
```diff
@@ -166,72 +166,94 @@ const processRouterDest = async (inputs, reqMetadata) => {
...
+
+  const singleEventArray = [];
+  const multipleEventsArray = [];
```
Comment (utsabc) [inline src/v0/destinations/iterable/transform.js:173]:
```suggestion
  const uniqueEventsPerUser = [];
  const multipleEventsPerUser = [];
```

Comment (mihir-4116) [reply src/v0/destinations/iterable/transform.js]: updated

File: src/v0/destinations/iterable/transform.js (hunk 3 above)
Comment (mihir-4116) [reply src/v0/destinations/iterable/transform.js]: updated

File: src/v0/destinations/iterable/transform.js (hunk 2 above)
Comment (mihir-4116) [reply src/v0/destinations/iterable/transform.js]: updated

**Review by chandumlg — APPROVED** [review]

**Pull request merged**